subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultLenientConfiguration.java

```diff
@@ -280,7 +280,7 @@ public ConfigurationInternal getConfiguration() {
     }
 
     public Set<ResolvedDependency> getFirstLevelModuleDependencies() {
-        return loadTransientGraphResults(getSelectedArtifacts()).getRootNode().getPublicView().getChildren();
+        return getFirstLevelModuleDependencies(Specs.SATISFIES_ALL);
```

Comment:
This change actually isn't required to make your new test pass, but I'm happy to leave it in as it simplifies things. Let's keep that in mind in case we hear about any performance regressions (I can't think of a non-performance reason that Adam would have done this in the first place.

Reply:
The performance test suite passed, so I guess it's not an issue. Or maybe this code predates the version with a spec?

Reply:
Possibly. I'm happier with the changed version, because it's a lot simpler and will be easier to maintain.